index.js

```diff
@@ -362,13 +367,80 @@ function filterFormatCodecs(format) {
     && (acodec !== 'none' || (acodec === 'none' && !protocol.includes('https')));
 }
 
-function filterFormatFps(format) {
+function filterVideoFormatFps(format) {
   const { fps, height } = format;
   return ((height >= 1080 && fps <= 30) || height < 1080);
 }
 
+function processAudioFormats(formats) {
+  // Filter out tracks that are not suitable (see comments below)
+  const filteredFormats = formats.filter((format) => filterAudioFormatCodecs(format));
+  // Sort the tracks because .find will return the first match
+  filteredFormats.sort(audioTrackSort)
+
+  return filteredFormats.length ? filteredFormats[0] : null;
+}
+
+function filterAudioFormatCodecs(format) {
+  const { acodec, audio_ext, abr, protocol, language } = format;
+
+  // Audio tracks that are m3u8 have audio_ext set, but acodec and abr are undefined. For some reason yt-dlp can't determine acodec and abr in these situations
+  if (acodec && acodec === 'none') {
+    // not an audio track
+    return false;
+  }
+
+  if (!acodec && !audio_ext && !abr) {
+    // not an audio track
+    return false;
+  }
+
+  // Tracks with protocol=https and acodec=mp4a are no good, because our gstreamer split playing pipeline
+  // can't seek on these tracks. I don't know why, seek succeeds but nothing happens.
+  if (protocol.includes('https') && acodec && acodec.includes('mp4a')) {
+    return false;
+  }
+
+  // If we know the track is non-english, skip it
+  // Not all youtube videos have language information in their audio tracks. I think youtube is just rolling this out now
+  if (language && !language.startsWith('en')) {
```

Comment:
We're probably gonna revisit this when we properly do i18n so not gonna worry too much about this. But I think doing a sort after the filter makes more sense? Having a non-english track is still better than nothing.

Reply:
done